app/views/pages/_extended_panel.html.erb

```diff
@@ -13,7 +13,12 @@
           </div>
         </div>
         <div class="col-xs-11" style="word-wrap: break-word;"><%= data.jobname %>
-          <div><%= data.pbsid %></div>
+          <% if page=='index' %>
+            <div><%= link_to data.pbsid, show_path(cluster: data.native_attribs.select{|attrib| attrib.name=="Cluster"}[0].value.downcase , pbsid: data.pbsid.to_s) %></div>
+          <% elsif page=='show' %>
+             <div><%= data.pbsid %></div>
+             <div><%= @data.queue.capitalize %></div>
+          <% end %>
```

Comment:
How about we just have the same view in both cases with the queue to the right of the id and and move this above the job name.

```
<%= "#{data.pbsid} - #{data.queue}" %><br>
<%= data.jobname %>
```

(br or divs - whatever produces the desired result)

Reply:
If you want the link to be to the show page I don't see a problem with having the link appear in both index and show pages. So you can get rid of the extra logic.